src/stores/FiltersHistoryStore.ts

```diff
@@ -0,0 +1,76 @@
+/** ****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+
+import { observable, computed, action } from 'mobx';
+import { IndexedDB, IndexedDbStores, indexedDbLimits } from '../api/indexedDb';
+import { SearchPanelType } from '../components/search-panel/SearchPanel';
+import { FiltersToSave } from './FilterAutocompletesStore';
+
+export interface FiltersHistory {
+	timestamp: number;
+	type: SearchPanelType;
+	filters: FiltersToSave;
+}
+
+class FiltersHistoryStore {
+	constructor(private indexedDb: IndexedDB) {
+		this.init();
+	}
+
+	@observable
+	public history: FiltersHistory[] = [];
+
+	@action
+	public addHistoryItem = async (newFilters: FiltersHistory) => {
+		const hasSame = this.history.some(({ filters }) => {
+			return JSON.stringify(filters) === JSON.stringify(newFilters.filters);
+		});
+		if (hasSame) {
+			return;
+		}
+		if (this.isFull) {
+			// eslint-disable-next-line @typescript-eslint/no-unused-vars
+			const [_, ...rest] = this.history;
```

Comment:
This can be rewritten without disabling eslint rule
something like `this.history.slice(1, this.history.length)`